src/GuiBase/Utils/KeyMappingManager.cpp

```diff
@@ -319,47 +379,174 @@ KeyMappingManager::~KeyMappingManager() {
 
 std::string KeyMappingManager::enumNamesFromMouseButtons( const Qt::MouseButtons& buttons ) {
     std::string returnText;
+    std::string sep = "";
     if ( buttons == Qt::NoButton ) return "NoButton";
-    if ( buttons & Qt::LeftButton ) returnText += "LeftButton ";
-    if ( buttons & Qt::RightButton ) returnText += "RightButton ";
-    if ( buttons & Qt::MiddleButton ) returnText += "MiddleButton ";
-    if ( buttons & Qt::BackButton ) returnText += "BackButton ";
-    if ( buttons & Qt::ForwardButton ) returnText += "ForwardButton ";
-    if ( buttons & Qt::TaskButton ) returnText += "TaskButton ";
-    if ( buttons & Qt::ExtraButton4 ) returnText += "ExtraButton4 ";
-    if ( buttons & Qt::ExtraButton5 ) returnText += "ExtraButton5 ";
-    if ( buttons & Qt::ExtraButton6 ) returnText += "ExtraButton6 ";
-    if ( buttons & Qt::ExtraButton7 ) returnText += "ExtraButton7 ";
-    if ( buttons & Qt::ExtraButton8 ) returnText += "ExtraButton8 ";
-    if ( buttons & Qt::ExtraButton9 ) returnText += "ExtraButton9 ";
-    if ( buttons & Qt::ExtraButton10 ) returnText += "ExtraButton10 ";
-    if ( buttons & Qt::ExtraButton11 ) returnText += "ExtraButton11 ";
-    if ( buttons & Qt::ExtraButton12 ) returnText += "ExtraButton12 ";
-    if ( buttons & Qt::ExtraButton13 ) returnText += "ExtraButton13 ";
-    if ( buttons & Qt::ExtraButton14 ) returnText += "ExtraButton14 ";
-    if ( buttons & Qt::ExtraButton15 ) returnText += "ExtraButton15 ";
-    if ( buttons & Qt::ExtraButton16 ) returnText += "ExtraButton16 ";
-    if ( buttons & Qt::ExtraButton17 ) returnText += "ExtraButton17 ";
-    if ( buttons & Qt::ExtraButton18 ) returnText += "ExtraButton18 ";
-    if ( buttons & Qt::ExtraButton19 ) returnText += "ExtraButton19 ";
-    if ( buttons & Qt::ExtraButton20 ) returnText += "ExtraButton20 ";
-    if ( buttons & Qt::ExtraButton21 ) returnText += "ExtraButton21 ";
-    if ( buttons & Qt::ExtraButton22 ) returnText += "ExtraButton22 ";
-    if ( buttons & Qt::ExtraButton23 ) returnText += "ExtraButton23 ";
-    if ( buttons & Qt::ExtraButton24 ) returnText += "ExtraButton24 ";
+    if ( buttons & Qt::LeftButton )
```

Comment:
Couldn't we use a macro here, to avoid all this duplicated code ?
(I can try to find time to write it ..)

Reply:
not necessary, next time ?

Reply:
done.